From 3f91eae3bd0a4867cf954ca8a08cd207f9e22dfd Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 18 May 2026 10:01:53 +1000
Subject: [PATCH] uiautomationcore: Add UiaDisconnectAllProviders stub.

(cherry picked from commit 1d6e243ff6872a34972e8f9c7d1cfd569ed7961e)

CW-Bug-Id: #27696
---
 dlls/uiautomationcore/uia_provider.c        | 7 +++++++
 dlls/uiautomationcore/uiautomationcore.spec | 2 +-
 include/uiautomationcoreapi.h               | 1 +
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/dlls/uiautomationcore/uia_provider.c b/dlls/uiautomationcore/uia_provider.c
index ce4aa0b53ae..ca3de7e6ce0 100644
--- a/dlls/uiautomationcore/uia_provider.c
+++ b/dlls/uiautomationcore/uia_provider.c
@@ -2338,3 +2338,10 @@ HRESULT WINAPI UiaDisconnectProvider(IRawElementProviderSimple *elprov)
 
     return S_OK;
 }
+
+HRESULT WINAPI UiaDisconnectAllProviders(void)
+{
+    FIXME("stub\n");
+
+    return S_OK;
+}
diff --git a/dlls/uiautomationcore/uiautomationcore.spec b/dlls/uiautomationcore/uiautomationcore.spec
index 69e1f0ba9f9..ebde5ea0c65 100644
--- a/dlls/uiautomationcore/uiautomationcore.spec
+++ b/dlls/uiautomationcore/uiautomationcore.spec
@@ -55,7 +55,7 @@
 @ stub TransformPattern_Rotate
 @ stdcall UiaAddEvent(ptr long ptr long ptr long ptr ptr)
 @ stdcall UiaClientsAreListening()
-#@ stub UiaDisconnectAllProviders
+@ stdcall UiaDisconnectAllProviders()
 @ stdcall UiaDisconnectProvider(ptr)
 @ stdcall UiaEventAddWindow(ptr long)
 @ stdcall UiaEventRemoveWindow(ptr long)
diff --git a/include/uiautomationcoreapi.h b/include/uiautomationcoreapi.h
index fbfbc46b09c..3316aff8a54 100644
--- a/include/uiautomationcoreapi.h
+++ b/include/uiautomationcoreapi.h
@@ -560,6 +560,7 @@ HRESULT WINAPI UiaEventAddWindow(HUIAEVENT huiaevent, HWND hwnd);
 HRESULT WINAPI UiaEventRemoveWindow(HUIAEVENT huiaevent, HWND hwnd);
 BOOL WINAPI UiaHasServerSideProvider(HWND hwnd);
 BOOL WINAPI UiaClientsAreListening(void);
+HRESULT WINAPI UiaDisconnectAllProviders(void);
 
 #ifdef __cplusplus
 }
